src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,253 @@
+        }
+    }
+
+    //prints specific task that is done
+    public static void setDoneTask(Task t, int i) throws IOException {
+        tasks.get(i).setDone(true);
+        writeToFile();
+
+        printHorizontalLine();
+        System.out.println("Nice, I've marked this task as done:");
+        String type = t.type;
+        String icon = t.getStatusIcon();
+        System.out.println("[" + type + "]" + " [" + icon + "] " + t.description);
+        printHorizontalLine();
+    }
+
+
+    public static void deleteTask (Task t){
+        tasks.remove(t);
+
+        printHorizontalLine();
+        System.out.println("Noted, I've removed this task:");
+        String type = t.type;
+        String icon = t.getStatusIcon();
+        System.out.println("[" + type + "]" + " [" + icon + "] " + t.description);
+        System.out.println("Now you have " + tasks.size() + " tasks in the list");
+        printHorizontalLine();
+    }
+
+    public static void printHorizontalLine() {
+        for (int i = 0; i < 11; i++) {
+            System.out.print("-");
+        }
+        System.out.println("-");
+    }
+
+    public static void printStartMessage() {
+        System.out.println("Hello! I'm Duke");
+        System.out.println("What can I do for you?");
+        printHorizontalLine();
+    }
+
+    public static void printByeMessage() {
+        printHorizontalLine();
+        System.out.println("Bye. Hope to see you again soon!");
+        printHorizontalLine();
+    }
+
+    public static void writeToFile() throws IOException {
+        FileWriter fw = new FileWriter(filePath, true);
+
+        for (Task task : tasks) {
+            try {
+                clearFile();
+
+                String status = "ND";
+                if (task.getStatusIcon().equals("X")) {
+                    status = "D";
+                }
+                fw.write(task.type + " | " + status + " | "
+                        + task.description + System.lineSeparator());
+            } catch (IOException e) {
+                System.out.println("Something went wrong: " + e.getMessage());
+            }
+        }
+        fw.close();
+    }
+
+    public static void clearFile() throws IOException {
+        FileWriter fw = new FileWriter(filePath);
+        fw.write("");
+        fw.close();
+    }
+
+    public static void readFile() throws FileNotFoundException {
+        Scanner s = new Scanner(taskText);
+        String[] arr;
+
+        while (s.hasNext()) {
+            String text = s.nextLine();
+            arr = text.split("\\|");
+
+            boolean status = arr[1].trim().equals("D");
+
+            Task t;
+            switch (arr[0].trim()) {
+            case "T":
+                t = new Todo(arr[2].trim());
+                t.isDone = status;
+                tasks.add(t);
+
+
+                break;
+            case "D": {
+                String fulltext = arr[2];
```

Comment:
Doesn't match the naming convention: variables should be in camelCase i.e.`fullText`